python/gsw/main.py

```diff
@@ -0,0 +1,350 @@
+                iridium_res = es.index(index='iridium_report', doc_type='report', body=ir_report)
+                # Print whether or not indexing was successful
+                print("Iridium Report Status: "+iridium_res['result']+"\n\n")
+
+    def disconnect(self):
+        '''
+        Stops the thread which is checking emails
+        and posting reports to elasticsearch
+        '''
+        self.run_email_thread=False
+        self.check_email_thread.join()
+
+# Get keys for connecting to email account and elasticsearch server
+try:
+    with open(os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../usb_console/configs/radio_keys.json')))as radio_keys_config_file:
+        radio_keys_config = json.load(radio_keys_config_file)
+    with open(os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../usb_console/configs/server_keys.json')))as server_keys_config_file:
+        server_keys_config = json.load(server_keys_config_file)
+except json.JSONDecodeError:
+    print("Could not load config files. Exiting.")
+    raise SystemExit
+except KeyError:
+    print("Malformed config file. Exiting.")
+    raise SystemExit
+
+# Create a read_iridium object
+readIr=read_iridium(radio_keys_config, server_keys_config)
+# Start checking emails and posting reports
+readIr.connect()
+
+# Set up the SwaggerUI API
+swagger_config={
+    "headers":[],
+    "specs":[
+        {
+            "endpoint":"apispec_1",
+            "route":"/apispec_1.json",
+            "rule_filter":lambda rule:True,
+            "model_filter":lambda tag:True
+        }
+    ],
+    "static_url_path": "/flassger_static",
+    "swagger_ui":True,
+    "specs_route":"/swagger/"
+}
+swagger=Swagger(app, config=swagger_config)
+
+# Endpoint for testing post requests
+@app.route("/test", methods=["POST"])
+@swag_from("endpoint_configs/echo_config.yml")
+def echo():
+    input_json=request.get_json()
+    message=str(input_json["message"])
+    res={"Recieved": message}
+    return json.dumps(res)
+
+# Endpoint for indexing data in elasticsearch
+# Mostly for testing purposes. We don't use this to actually index data in elasticsearch
+@app.route("/telemetry", methods=["POST"])
+@swag_from("endpoint_configs/telemetry_config.yml")
+def index_sf_report():
+    #connect to elasticsearch
+    es=Elasticsearch([{'host':readIr.es_server,'port':readIr.es_port}])
+
+    sf_report=request.get_json()
+    #index statefield report in elasticsearch
+    sf_res = es.index(index='statefield_report', doc_type='report', body=sf_report)
+    res={"Report Status": sf_res['result']}
+    return res
+
+# Endpoint for getting data from the statefield reports index in elasticsearch
+@app.route("/search-statefields", methods=["POST"])
+@swag_from("endpoint_configs/search_statefields_config.yml")
+def get_statefield():
+    # Connect to elasticsearch
+    es=Elasticsearch([{'host':readIr.es_server,'port':readIr.es_port}])
```

Comment:
This object is declared several times over the course of the file. Is this a heavy object to construct? If it is, consider moving it to the top of the file so that it can be shared across all of the code.